### ISSUE TITLE
C# -> VB: Type members in a module cannot be shared

### ISSUE THREAD

**Issue by RussellSmith2** [icsharpcode/CodeConverter#85]
### Input code
```C#
static partial class SpecializedCollections
{
        public static readonly byte[] EmptyBytes = EmptyArray<byte>();
        public static readonly object[] EmptyObjects = EmptyArray<object>();
}
```

### Erroneous output
```vbnet
Partial Module SpecializedCollections
        Public Shared ReadOnly EmptyBytes As Byte() = EmptyArray(Of Byte)()
        Public Shared ReadOnly EmptyObjects As Object() = EmptyArray(Of Object)()
End Module
```

### Expected output
```vbnet
Partial Module SpecializedCollections
        Public ReadOnly EmptyBytes As Byte() = EmptyArray(Of Byte)()
        Public ReadOnly EmptyObjects As Object() = EmptyArray(Of Object)()
End Module
```

### Details
The example shows fields, but applies to anything declared in a module.

**Comment by RussellSmith2** [comment]
Also, module constructors cannot be private.

**Comment by GrahamTheCoder** [comment]
The solution to this is a refactor to increase the number of TokenContexts to be the same as the C# -> VB converter. Then `IgnoreInContext` just needs an entry to ignore shared for `MemberInModule`

**Comment by thenoelman** [comment]
I'm interested in contributing to this project if you'd like to take a newbie on.  I'm very comfortable with C#.  I cloned your project and I'm trying to see figure things out.  Your documentation is a little light (possibly I could help with that too).  Please let me know how I can best help out!

**Comment by GrahamTheCoder** [comment]
Hi @thenoelman. Yep I'm very willing to help you get started. I've added a [Contributing.md](https://github.com/icsharpcode/CodeConverter/blob/master/.github/CONTRIBUTING.md) with what I was going to write in this message for general guidance.

If you ask questions on the issue you want to tackle (I'm assuming it's this one), I'll try to help, then if you think there's something that would help someone else you could perhaps document it somewhere.

A good test to copy for this would be: https://github.com/icsharpcode/CodeConverter/blob/master/Tests/VB/MemberTests.cs#L66
Then if you put a breakpoint in `IgnoreInContext` and debug the test you'll probably be able to see the issue. The method doesn't have enough information to differentiate between a method in a class and a method in a module. If you then compare to the `IgnoreInContext` for VB -> C# you can see I've increased the amount of contexts, which fixed some similar issues there.

**Comment by thenoelman** [comment]
@GrahamTheCoder thanks!  Starting here seems as good as any place.  I think that I was already in the right spot in the code when I asked the question.  You need this line, right?

![firstissue](https://user-images.githubusercontent.com/16550366/39611913-03674d18-4f10-11e8-8ab3-b8794624c034.jpg)

Where do I copy the test case to?  How do I best debug for this project.  I'm used to debugging with a U/I.

**Comment by GrahamTheCoder** [comment]
Yep that's the place. So you can see it always returns true, whereas it needs to depend on the context, like the one just below. However, the context doesn't currently contain enough information, so it needs to be beefed up a bit.

To debug, you want to run one of the tests in debug mode. If you open Test Runner Explorer and right click a test, there should be an option to run it in debug mode.

**Comment by thenoelman** [comment]
@GrahamTheCoder thanks for the assistance.  I'll certainly help to update the documentation as we work through this.  Sorry if I'm asking newbie questions, but I'm still getting up to speed with your project.  

I'm in debug mode.  
![2018-05-08_20-52-44](https://user-images.githubusercontent.com/16550366/39794962-eecfdb9a-5301-11e8-88ec-6d5b9d91a8a2.jpg)

If I put a breakpoint in next to the line of code that I added, and press the green button, I get this message:

![2018-05-08_20-54-57](https://user-images.githubusercontent.com/16550366/39794984-1b9a12ee-5302-11e8-9336-882d3efbe0cc.jpg)

Where do I set the startup project?  I'm not sure what you mean by "open Test Runner Explorer".  Where is that?  I'm the tests folder?  

I'll write up documentation for this as we work through it.  I realize that this is a trivial bug for someone familiar with the project, so I appreciate your patience.

Thanks!
